File: xwallpaper/src/main/java/vn/loitp/app/utilities/LLog.java
package vn.loitp.app.utilities;

import android.util.Log;

import vn.loitp.app.common.Constants;

public class LLog {
    public static void d(String tag, String msg) {
        if (Constants.IS_DEBUG) {
            Log.d(tag, msg);
        }
    }

    public static void e(String tag, String msg) {
        if (Constants.IS_DEBUG) {
            Log.e(tag, msg);
        }
    }
}

File: core/src/main/java/com/core/utilities/LLog.kt
package com.core.utilities

import android.util.Log
import kotlin.math.min

object LLog {

    private const val maxLogSize = 4000

    @JvmStatic
    fun d(tag: String, msg: String) {
        var i = 0
        while (i < msg.length) {
            Log.d(tag, msg.substring(i, min(a = msg.length, b = i + maxLogSize)))
            i += maxLogSize
        }
    }

    @JvmStatic
    fun e(tag: String, msg: String) {
        var i = 0
        while (i < msg.length) {
            Log.e(tag, msg.substring(i, min(a = msg.length, b = i + maxLogSize)))
            i += maxLogSize
        }
    }

    @JvmStatic
    fun i(tag: String, msg: String) {
        var i = 0
        while (i < msg.length) {
            Log.i(tag, msg.substring(i, min(a = msg.length, b = i + maxLogSize)))
            i += maxLogSize
        }
    }
}
